game/plugin/skills/herblore/src/herblore.plugin.kts

```diff
@@ -0,0 +1,33 @@
+import org.apollo.game.message.impl.ItemOptionMessage
+import org.apollo.game.message.impl.ItemOnItemMessage
+
+on { ItemOptionMessage::class }
+        .where { option == 1 }
+        .then {
+            val unidentified = Herb[id] ?: return@then
+            it.startAction(IdentifyHerbAction(it, slot, unidentified))
+            terminate()
+        }
+
+on { ItemOnItemMessage::class }
+        .where { (id == PESTLE_MORTAR || id == KNIFE) && (targetId == PESTLE_MORTAR || targetId == KNIFE) }
+        .then {
+            val raw = CrushIngredient[id] ?: return@then
+            it.startAction(CrushIngredientAction(it, raw))
+            terminate()
+        }
+
+on { ItemOnItemMessage::class }
+        . where { id == WATER_VIAL || targetId == WATER_VIAL }
+        . then {
+            val unfinished = UnfinishedPotion[id] ?: return@then
+            it.startAction(MakeUnFinishedPotionAction(it, unfinished))
+            terminate()
+        }
+
+on { ItemOnItemMessage::class }
+        . then {
+            val finished = FinishedPotion[Pair(targetId, id)] ?: return@then
```

Comment:
You need to check which way round the `id` and `targetId` should go here - it needs to be `unfinished` first, `ingredient` second, so you should check which is which.